lib/rules/no-child-content.js

```diff
@@ -0,0 +1,157 @@
+  return node.type === 'VText' && node.value.trim() === ''
+}
+
+/**
+ * @param {Position} pos1
+ * @param {Position} pos2
+ * @returns {'less' | 'equal' | 'greater'}
+ */
+function comparePositions(pos1, pos2) {
+  if (
+    pos1.line < pos2.line ||
+    (pos1.line === pos2.line && pos1.column < pos2.column)
+  ) {
+    return 'less'
+  }
+
+  if (
+    pos1.line > pos2.line ||
+    (pos1.line === pos2.line && pos1.column > pos2.column)
+  ) {
+    return 'greater'
+  }
+
+  return 'equal'
+}
+
+/**
+ * @param {(VNode | Token)[]} nodes
+ * @returns {SourceLocation | undefined}
+ */
+function getLocationRange(nodes) {
+  /** @type {Position | undefined} */
+  let start
+  /** @type {Position | undefined} */
+  let end
+
+  for (const node of nodes) {
+    if (!start || comparePositions(node.loc.start, start) === 'less') {
+      start = node.loc.start
+    }
+
+    if (!end || comparePositions(node.loc.end, end) === 'greater') {
+      end = node.loc.end
+    }
+  }
+
+  if (start === undefined || end === undefined) {
+    return undefined
+  }
+
+  return { start, end }
+}
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    hasSuggestions: true,
+    type: 'problem',
+    docs: {
+      description:
+        "disallow element's child contents which would be overwritten by a directive like `v-html` or `v-text`",
+      categories: undefined,
+      url: 'https://eslint.vuejs.org/rules/no-child-content.html'
+    },
+    fixable: null,
+    schema: [
+      {
+        type: 'object',
+        additionalProperties: false,
+        properties: {
+          additionalDirectives: {
+            type: 'array',
+            uniqueItems: true,
+            minItems: 1,
+            items: {
+              type: 'string'
+            }
+          }
+        },
+        required: ['additionalDirectives']
+      }
+    ]
+  },
+  /** @param {RuleContext} context */
+  create(context) {
+    const directives = new Set(['html', 'text'])
+
+    /** @type {RuleOption | undefined} */
+    const option = context.options[0]
+    if (option !== undefined) {
+      for (const directive of option.additionalDirectives) {
+        directives.add(directive)
+      }
+    }
+
+    const templateBody = context.getSourceCode().ast.templateBody
+    const templateComments = templateBody ? templateBody.comments : []
+
+    return defineTemplateBodyVisitor(context, {
+      /** @param {VDirective} directiveNode */
+      'VAttribute[directive=true]'(directiveNode) {
+        const directiveName = directiveNode.key.name.name
+        const elementNode = directiveNode.parent.parent
+
+        const elementComments = templateComments.filter((comment) =>
+          inRange(elementNode.range, comment)
+        )
+
+        const childNodes = [...elementNode.children, ...elementComments]
```

Comment:
I think you can use the TokenStore to extract all of the content tokens as follows:

```js
const tokenStore = context.parserServices.getTemplateBodyTokenStore()
const childContentTokens = tokenStore.getTokensBetween(elementNode.startTag, elementNode.endTag, { includeComments: true, filter:token => token.type !== 'HTMLWhitespace' })
```

Reply:
I simplified the retrieval of comments with this function. However, completely replacing the logic with the tokens returned by this function seemed more effort than it's worth (the suggested fix then doesn't remove whitespace e.g. in `{{ foo }}` anymore). See [`9566ee4`](https://github.com/vuejs/eslint-plugin-vue/pull/1707/commits/9566ee4a20fffdb13feca7cf7c5db58529971504).

Reply:
I think your method is better. thank you!